server/src/controllers/users.js

```diff
@@ -73,13 +73,18 @@ export const createUser = async (req, res, next) => {
 export const updateUser = async (req, res, next) => {
   try {
     const id = req.params.id
-    const updates = req.body
+    const { username, email, calorieLimit, image } = req.body
 
     if (id !== req.user.id && [ROLES.User].includes(req.user.role)) {
       return res.status(403).json({})
     } else {
-      await Users.findByIdAndUpdate(id, updates, { new: true })
-      res.status(200).json({})
+      const updatedUser = await Users.findByIdAndUpdate(
+        id,
+        { username, email, calorieLimit, image },
+        { new: true }
```

Comment:
new რატო? თუ არ არსებობს ასეთი იუზერი 400 უნდა დააბრუნო თავისი მესიჯით

Reply:
"By default, findOneAndUpdate() returns the document as it was before update was applied. If you set new: true, findOneAndUpdate() will instead give you the object after update was applied."

ეს პროსტო დააფდეითებულ იუზერს არ აბუნებს? ვერ ვხვდები აქ რაარი პრობლემა.